nucypher/policy/conditions/lingo.py

```diff
@@ -104,21 +105,46 @@ def __init__(self, comparator: str, value: Any):
 
         self.comparator = comparator
         self.value = value
+        self.key = key
 
     def _sanitize_value(self, value):
         try:
             return ast.literal_eval(str(value))
         except Exception:
             raise self.InvalidExpression(f'"{value}" is not a permitted value.')
 
+    def _process_data(self, data: Any) -> Any:
+        if self.key and isinstance(data, dict):
+            try:
+                processed_data = data[self.key]
+            except KeyError:
+                raise KeyError(f"Key '{self.key}' not found in return data.")
+        elif isinstance(self.key, int) and isinstance(data, (list, tuple)):
+            try:
+                processed_data = data[self.key]
+            except IndexError:
+                raise IndexError(f"Index '{self.key}' not found in return data.")
+        else:
+            processed_data = data
+
+        return processed_data
+
     def eval(self, data) -> bool:
         if is_context_variable(self.value):
             # programming error if we get here
             raise RuntimeError(
                 f"'{self.value}' is an unprocessed context variable and is not valid "
                 f"for condition evaluation."
             )
-        left_operand = self._sanitize_value(data)
+        if is_context_variable(self.key):
+            # programming error if we get here
+            raise RuntimeError(
+                f"'{self.key}' is an unprocessed context variable and is not valid "
+                f"for condition evaluation."
+            )
```

Comment:
Even though the error message will be less descriptive, it's possible to combine these two checks into check with a more generic message.  Considering this is a "Programming Error" that is not intended as part of normal flow-control I think it's worth the trade-off.